content/en/apps/tutorials/application-tests.md

```diff
@@ -0,0 +1,600 @@
+---
+
+{{% pageinfo %}}
+This tutorial takes you through testing the various configurable components of CHT applications using [`cht-conf-test-harness`](http://docs.communityhealthtoolkit.org/cht-conf-test-harness/).
+
+{{% /pageinfo %}}
+
+## Prerequisites
+
+* Complete these tutorials:
+  - [Building App Forms]({{< ref "apps/tutorials/app-forms" >}})
+  - [Building A Simple Task]({{< ref "apps/tutorials/tasks-1" >}})
+  - [Building Target Widgets]({{< ref "apps/tutorials/targets" >}})
+  - [Building Contact Summary]({{< ref "apps/tutorials/contact-summary" >}})
+
+## CHT Application Testing
+
+---
+
+CHT applications are greatly configurable. Depending on the number and the complexity of the app components, it can take huge time and effort to test them manually. Some components, such as tasks, behave differently over time and are particularly difficult to test. As the project evolves, the configuration is often updated with new components, and changes are made to the existing components. After each change, you need to test not only the new components but also the old ones to make sure that the app works correctly. To facilitate the testing process, app builders are encouraged to write automated tests for their app using [cht-conf-test-harness](http://docs.communityhealthtoolkit.org/cht-conf-test-harness/).
+
+Using `cht-conf-test-harness` (also simply referred as `harness`), you can write tests and run them with [Mocha](https://mochajs.org/) testing framework to test the behavior of different components in a CHT application. You can use the [chai library](https://www.chaijs.com/) or other assertion libraries.
+
+## Preparing
+
+Writing tests for CHT apps requires a good understanding of the project workflows and requirements. It is encouraged that the tests be written as part of the app-building process. To test using the harness, there are a few things you need to set up first:
+1. From the previous tutorials, you should have a [functioning CHT instance with `cht-conf` installed locally]({{< ref "apps/tutorials/local-setup" >}}) and a [project folder set up]({{< ref "apps/tutorials/local-setup#3-create-and-upload-a-blank-project" >}}).
+2. If your `package.json` file does not already have them, add `cht-conf-test-harness`, `chai`, and `mocha` by running this in your command-line:
+    ```shell
+    npm install --save cht-conf-test-harness chai mocha
+    ```
+3. Also, add the following scripts to `package.json`, if not already present:
+
+    ```json
+    "scripts": {
+        "test": "npm run compile-forms && npm run compile-app-settings && npm run unittest",
+        "compile-app-settings": "npx cht --no-check compile-app-settings",
+        "compile-forms": "npx cht --no-check convert-app-forms convert-contact-forms",
+        "unittest": "mocha test/**/*.spec.js --timeout 20000"
+      }
+    ```
+    After adding these scripts, you are able to run the tests by running one of these commands from the command-line:
+
+    <table>
```

Comment:
I think table can be replaced with below structure (Reference: https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#use-markdown-notation)
|Command|Description|
|---|---|
|npm run test|Compiles the app settings, converts the forms, then runs the tests. <sup>[recommended]|
|npm run unittest|Only runs the tests.|

Reply:
I used HTML table because the `npm run unittest` was wrapping. But that might not be an enough reason. I've changed it to markdown. Thanks.

Reply:
Also, if you know a way to not wrap it, please let me know. It is looking like this:
<img width="249" alt="image" src="https://user-images.githubusercontent.com/20041732/183677982-500bafed-eb53-4411-80d6-77e6be393309.png">